lib/src/test/resources/compileContracts.sh

```diff
@@ -1,47 +1,113 @@
 #!/bin/bash
 
-pushd "$(dirname "$0")" || exit
-mkdir -p "compiled_v0"
+V1_COMPILER_BUILD_PATH_VAR="V1_COMPILER_BUILD_PATH"
+V2_COMPILER_BUILD_PATH_VAR="V2_COMPILER_BUILD_PATH"
```

Comment:
what does this do?

Reply:
Those are just names of env variables we are retrieving actual paths from.
1. I thought it would be good to have a top level declaration that specifies the names should they be changed, hence `_VAR` postfix.
2. We can later reference those names when throwing error.

Reply:
they probably won't, and even if so - it won't hurt us in any way, as this script is just our helper and nothing user-facing

Reply:
I know it probably won't, but I don't see how having them up top is bad as well. Having those names defined also helps with printing error reason with `$COMPILER_BUILD_PATH_VAR`.

Or do you have some specific change in mind?

Reply:
imho the more unnecessary code the worse. you can achieve exactly the same thing by just checking the variables existence directly and assigning one or the other to your general variable. the error should be then moved to this place and the script should fail fast if none of those is defined. wdyt?

Reply:
I was basing this on a convention to declare constants and export env variables at the top level. Since we also need to specifically reference the name of those env variables to print an error message, at top levels we could instead have names of those variables (to avoid situation where debug info is not accurate), not their values.

I don't mind the approach you suggest though, I thought of it in the first place but then replaced it based on convention. If you don't find this convincing, I will change it.

Reply:
Sure I get this, and appreciate your idea, but I think this approach is more sensible when you actually declare some constants and not try to declare env variables. My understanding is that those variables have to be exported beforehand, so I think this just brings a bit of confusion.

so to sum up, so we're on the same page, my proposition is to:
- at the top of the script have a check for both variables
- if one of them (or maybe even both? I think we could allow this - what do you think?) is set, we can continue
- if none of them is set we print out an error that there are no cairo binaries and variables X and/or Y are not set

sounds good?

Reply:
Sure.
As for compilers, they require different flags, so maybe we should require both to be set. I'm not sure v2 compilers compiles v1 contracts and v1 definitely won't compile v2 contracts.